.github/workflows/scripts/scanner-fetch-nvd.sh

```diff
@@ -0,0 +1,121 @@
+
+    retry_intervals=(2 4 8) # Array of retry intervals in seconds
+    success=false
+
+    for i in "${retry_intervals[@]}"; do
+        if curl -s --fail --show-error -H "apiKey: $API_KEY" "$base_url?pubStartDate=$period_start_date&pubEndDate=$period_end_date&startIndex=$current_start_index" -o "$dir_name/$file_name"; then
+            success=true
+            break # Exit
+        else
+            sleep "$i" # Wait for next retry
+        fi
+    done
+
+    if [ "$success" = false ]; then
+        echo "Failed after retries"
+        exit 1
+    fi
+
+    if [ ! -s "$dir_name/$file_name" ] || ! jq empty "$dir_name/$file_name"; then
+      echo "Downloaded file is empty or invalid for the period: $period_start_date to $period_end_date, starting index $current_start_index"
+      exit 1
+    fi
+
+    total_results=$(jq '.totalResults' "$dir_name/$file_name") || { echo "jq command failed"; exit 1; }
+
+    current_start_index=$((current_start_index + results_per_page))
+    sleep 2
+  done
+}
+
+
+mkdir "$dir_name"
+
+# Function to fetch data for each year
+fetch_data_for_year() {
+    local year=$1
+    local dir_name=$2
+    local periods=(
+        "${year}-01-01T00:00:00.000 ${year}-03-31T23:59:59.999"
+        "${year}-04-01T00:00:00.000 ${year}-06-30T23:59:59.999"
+        "${year}-07-01T00:00:00.000 ${year}-09-30T23:59:59.999"
+        "${year}-10-01T00:00:00.000 ${year}-12-31T23:59:59.999"
+    )
+
+    for period in "${periods[@]}"; do
+        local START_DATE
+        START_DATE=$(echo "$period" | cut -d ' ' -f1)
+
+        local END_DATE
+        END_DATE=$(echo "$period" | cut -d ' ' -f2)
+
+        echo "Fetching data from ${START_DATE} to ${END_DATE}"
+        fetch_data_for_period "${START_DATE}" "${END_DATE}" "${dir_name}"
+        sleep 2
+    done
+}
+
+# Function to process and combine data for each year
+process_year_data() {
+    local year=${1}
+    local dir_name=${2}
+    local file_count
+    file_count=$(find "${dir_name}" -maxdepth 1 -type f -name "*${year}*.json" | wc -l)
+    echo "There are ${file_count} files in the ${dir_name} directory for year ${year}."
+
+    if [ "${file_count}" -gt 0 ]; then
+        if jq -cs '{ vulnerabilities: map(.vulnerabilities) | add }' "${dir_name}"/*"${year}"*.json > "${dir_name}/combined_${year}.json"; then
+            echo "The consolidated file size for ${year} is $(stat -c %s "${dir_name}/combined_${year}.json") bytes."
+            find "${dir_name}" -maxdepth 1 -type f ! -name 'combined_*.json' -exec rm {} \;
+        else
+            echo "Failed to combine JSON files for ${year}."
+            exit 1
+        fi
+    else
+        echo "No files to combine for year ${year}."
+    fi
+}
+
+# Function to compress and upload data
+compress_and_upload() {
+    local dir_name=${1}
+    if tar -czf "${dir_name}.tar.gz" -C "${dir_name}" .; then
+        echo "The size of the compressed file ${dir_name}.tar.gz is $(stat -c %s "${dir_name}.tar.gz") bytes."
```

Comment:
I've been unable to find any information about `stat -c`. Do you have a link to docs for it? Also, is itt possible to show this in MB? Same for the other calls to `stat`

Reply:
I am thinking to remove the file size log since we can always see the size in GCS bucket